libs/auth/src/java_card.ts

```diff
@@ -103,6 +127,16 @@ const GENERIC_STORAGE_SPACE = {
   ],
 } as const;
 
+/**
+ * Additional constructor inputs that only VxAdmin needs to provide, for card programming
+ */
+interface VxAdminParams {
+  vxAdminCertAuthorityCertPath: string;
+  vxAdminOpensslConfigPath: string;
+  vxAdminPrivateKeyPassword: string;
+  vxAdminPrivateKeyPath: string;
```

Comment:
eventually this is going to be based on the TPM-held vxadmin private key, right? That won't have a password, and we have to look into how OpenSSL can use the TPM.

But, that's a next PR I imagine.

Reply:
Really helpful reminder! This seems promising: https://github.com/tpm2-software/tpm2-tss-engine

This is going to be hard for me to test without real hardware, but lucky for us, I have a Lenovo ThinkPad on the way 😄

For VM development on Macs, we'll probably still want to support an old-school private key file and password. But I'll think of a way to abstract that away (in follow-up PRs)

Reply:
worth a chat about how we architect this maybe? I was thinking the TPM-based signature, since we have to do it in a number of different cases, might be provided as a library that calls out into a configurable system executable (e.g. `/vx/code/functions/sign.sh`).

Reply:
Ah good point re the need to perform TPM-based signatures in other contexts. Threw some time on the calendar for tomorrow morning - feel free to move around!